### PR TITLE
feat: an api test added. It evaluates inserting base64 encoded image

### PR DESCRIPTION
**Pull request by rezakarbasi**
…ded input or not

## Tickets

https://github.com/serengil/deepface/issues/XXX

### What has been done

With this PR, ...

## How to test

```shell
make lint && make test
```

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -77,6 +79,33 @@ def test_represent(self):
 
         logger.info("✅ representation api test is done")
 
+    def test_represent_encoded(self):
+        with open(image_path, "rb") as image_file:
```
Comment (serengil) [inline tests/test_api.py:83]: what if image_path?

you may consider to set it to "dataset/img1.jpg"

File: tests/test_api.py [hunk 2]
```diff
@@ -77,6 +79,33 @@ def test_represent(self):
...
+        assert response.status_code == 200
+        result = response.json
+        logger.debug(result)
+        assert result.get("results") is not None
+        assert isinstance(result["results"], list) is True
+        assert len(result["results"]) > 0
+        for i in result["results"]:
+            assert i.get("embedding") is not None
+            assert isinstance(i.get("embedding"), list) is True
+            assert len(i.get("embedding")) == 4096
```
Comment (serengil) [inline tests/test_api.py:103]: Facenet model creates 128 dimensional embeddings, not 4096

File: tests/test_api.py [hunk 3]
```diff
@@ -77,6 +79,33 @@ def test_represent(self):
 
         logger.info("✅ representation api test is done")
 
+    def test_represent_encoded(self):
```
Comment (serengil) [inline tests/test_api.py:82]: you may consider to add similar test case for numpy array input

File: tests/test_api.py (hunk 1 above)
Comment (rezakarbasi) [reply tests/test_api.py:83]: Sure that's the main problem

File: tests/test_api.py (hunk 3 above)
Comment (rezakarbasi) [reply tests/test_api.py:81]: Sure, I'll write it till end of the day

Comment (serengil) [reply tests/test_api.py:81]: no problem, you can create a new PR - I am approving this

**Comment by serengil** [comment]
LGTM

**Pull request merged**